### PR TITLE
Fixed authRequired Middleware & Got API for Admin Dashboard Working 

### PR DESCRIPTION
**Pull request by ahmedseragcodes**
#Loom Video Link
https://www.loom.com/share/c7ff9e3403cb449b9bb2291f98db475d

##Trello Link
https://trello.com/c/W49eewfD

# Description
Reformatted auth header getting passed through to oktaValidation in our authRequired middleware on the backend so that bearer string is separated from the actual token and actual token is what gets passed/verified. Fixed bug with doubly nested async await. Admin dashboard API now working, no longer getting errors for JWT cannot be parsed. Tesed with Swagged & Postman. Collaborated with Luis (Backend Engineer) and Ryan (Engineering Manager).

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)

## Change Status
- [ ] Complete, tested, ready to review and merge, but needs unit tests

# Checklist
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] My code has been reviewed by at least one peer
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] There are no merge conflicts

### CODE REVIEW

File: api/middleware/authRequired.js [hunk 1]
```diff
@@ -7,18 +7,22 @@ const oktaJwtVerifier = new OktaJwtVerifier(oktaVerifierConfig.config);
...
-  try {
-    const authHeader = req.headers.authorization || '';
-    oktaJwtVerifier
-      .verifyAccessToken(authHeader, oktaVerifierConfig.expectedAudience)
-      .then(async () => {
-        next();
-      })
-      .catch(next);
-  } catch (err) {
-    next(createError(401, err.message));
```
Comment (lgarcia2689) [inline api/middleware/authRequired.js:20]: Was able to verify that this piece of middleware data was not getting JSON WEB TOKEN from okta, thus not going through JWT Verification. With corrected code Web token is being passed and verified when calling on dashboard/incidents end points and passing correct response.

File: api/middleware/authRequired.js [hunk 2]
```diff
@@ -7,18 +7,22 @@ const oktaJwtVerifier = new OktaJwtVerifier(oktaVerifierConfig.config);
...
+  // eslint-disable-next-line prettier/prettier
+  oktaJwtVerifier
+    .verifyAccessToken(idToken, oktaVerifierConfig.expectedAudience)
+    .then(() => {
+      next();
+    })
+    .catch((err) => {
+      next(createError(401, err.message));
+    });
 };
```
Comment (lgarcia2689) [inline api/middleware/authRequired.js:26]: Great code Ahmed, this middleware is now working as intended calling and passing correct data. Not giving access to a user with an invalid or no token as this middle is supposed to do. Beautiful piece of code here

**Review by lgarcia2689 — APPROVED** [review]
Great work, overall code deleted needed to be removed due to no working correctly, and the code added is functional without errors after testing

File: api/middleware/authRequired.js (hunk 1 above)
Comment (ooladuwa) [reply api/middleware/authRequired.js:20]: This backend middleware was not operating in the desired manner. It failed to properly verify and pass the JWT. This code remedies that. The match/bearer line where you use regex was really interesting. My visual side would have liked for you to have shown what match returns. I think this was a really great job of isolating an issue, digging into it, and emerging with a seemingly solid solution.

**Review by ooladuwa — APPROVED** [review]
IMO you all did a great job identifying and isolating the problem and came up with a rather elegant solution.

**Review by paulstgermain — APPROVED** [review]

**Pull request merged**